src/main/java/org/openjdk/jextract/impl/IncludeHelper.java

```diff
@@ -82,12 +84,35 @@ static IncludeKind fromScoped(Declaration.Scoped scoped) {
     }
 
     private final EnumMap<IncludeKind, Set<String>> includesSymbolNamesByKind = new EnumMap<>(IncludeKind.class);
+
+    private final Map<String, Set<String>> propertiesByName = new HashMap<>();
+
     private final Set<Declaration> usedDeclarations = new HashSet<>();
     public String dumpIncludesFile;
 
+    /**
+     * Register an include, may be:Name or Name,flag1,flag2 orName,prop=value,flag
+     */
     public void addSymbol(IncludeKind kind, String symbolName) {
-        Set<String> names = includesSymbolNamesByKind.computeIfAbsent(kind, (_unused) -> new HashSet<>());
-        names.add(symbolName);
+        String[] parts = symbolName.split(",", 2);
+        String name = parts[0].trim();
+
+        includesSymbolNamesByKind.computeIfAbsent(kind, k -> new HashSet<>())
+                .add(name);
+
+        if (parts.length == 2) {
+            Set<String> props = propertiesByName
+                    .computeIfAbsent(name, k -> new HashSet<>());
+            for (String tok : parts[1].split(",")) {
+                props.add(tok.trim());
+            }
+        }
+
+        if (kind == IncludeKind.STRUCT || kind == IncludeKind.UNION) {
```

Comment:
Not sure I get this logic -- it seems like we're always setting a property P for a typedef with name N whenever we see it set for a struct N? That seems odd.

But in example as the one you added in the test are a bit ambiguous, as the same declaration induces both an anonymous struct decl and a typedef, so it is less clear which is which. If I write:

```
typedef struct Foo {
int a;
};
```

It seems to be that jextract only retains a "struct" tree, and throws away the typedef. In fact I was surprised that your test worked at all, as in the above case `dump-includes` would include something like this:

```
--include-struct Foo   # header: ....
```

And *not*:

```
--include-typedef Foo   # header: ....
```

So adding a property on the typedef should have no effect?